Makefile

```diff
@@ -1,34 +1,203 @@
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##@ Development
+
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+vet: ## Run go vet against code.
+	go vet ./...
+
+test-all: manifests generate fmt vet ## Run all tests.
+	go test ./... -coverprofile cover.out
+
+test-e2e: manifests generate fmt vet ## Run e2e tests.
 	. ./scripts/run_e2e_tests.sh
 
-.PHONY: gomod_tidy
-gomod_tidy:
-	go mod tidy
+test: manifests generate fmt vet ## Run unit tests.
+	go test `go list ./... | grep -v test` -coverprofile cover.out
+
+##@ Build
+
+build: generate fmt vet ## Build manager binary.
+	go build -o bin/manager main.go
+
+run: manifests generate fmt vet ## Run a controller from your host.
+	go run ./main.go
+
+docker-build: test ## Build docker image with the manager.
+	docker build -t ${IMG} .
+
+docker-push: ## Push docker image with the manager.
+	docker push ${IMG}
+
+##@ Deployment
+
+install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
+
+uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/crd | kubectl delete -f -
+
+deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/default | kubectl apply -f -
+
+undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/default | kubectl delete -f -
+
+
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+kustomize: ## Download kustomize locally if necessary.
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+rm -rf $$TMP_DIR ;\
+}
+endef
+
+.PHONY: bundle
```

Comment:
I'd expect `make bundle` to generate the same bundle/manifests but there are changed files.
```
        modified:   bundle/manifests/gitops-operator.clusterserviceversion.yaml
        modified:   config/manager/kustomization.yaml
        modified:   config/manifests/bases/gitops-operator.clusterserviceversion.yaml
```        

Also, we should get a clean validation ressults but there are some warnings
```
/home/wtam/go/src/github.com/redhat-developer/gitops-operator/bin/kustomize build config/manifests | operator-sdk generate bundle -q --overwrite --version 0.0.3  
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
INFO[0001] Creating bundle.Dockerfile                   
INFO[0001] Creating bundle/metadata/annotations.yaml    
INFO[0001] Bundle metadata generated suceessfully       
operator-sdk bundle validate ./bundle
INFO[0000] Found annotations file                        bundle-dir=bundle container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=bundle container-tool=docker
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=Application: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=ApplicationSet: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=AppProject: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=ArgoCD: provided API should have an example annotation 
INFO[0000] All validation tests have completed successfully 
```

Reply:
I am hitting a failure in execution. What is the recommended operator-sdk version ? I am using operator-sdk master branch latest commit 
```
$ operator-sdk version
operator-sdk version: "v1.11.0-1-gdfcc5751", commit: "dfcc57516b33abe8bc6b38f87748254c92b16eb4", kubernetes version: "v1.20.2", go version: "go1.16.6", GOOS: "darwin", GOARCH: "amd64"


$ make bundle
/Users/amit/go/src/github.com/redhat-developer/gitops-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
operator-sdk generate kustomize manifests -q
cd config/manager && /Users/amit/go/src/github.com/redhat-developer/gitops-operator/bin/kustomize edit set image controller=controller:latest
/Users/amit/go/src/github.com/redhat-developer/gitops-operator/bin/kustomize build config/manifests | operator-sdk generate bundle -q --overwrite --version 0.0.3  
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0001] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
INFO[0001] Creating bundle.Dockerfile                   
INFO[0001] Creating bundle/metadata/annotations.yaml    
INFO[0001] Bundle metadata generated suceessfully       
operator-sdk bundle validate ./bundle
ERRO[0000] Error: Value gitops-operator: invalid service account found in bundle. sa name cannot match service account defined for deployment spec in CSV 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=Application: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=ApplicationSet: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=AppProject: provided API should have an example annotation 
WARN[0000] Warning: Value argoproj.io/v1alpha1, Kind=ArgoCD: provided API should have an example annotation 
make: *** [bundle] Error 1

Reply:
@amitkrout I used v1.8 for testing locally. But it should work fine for v1.x. Let me check

Reply:
@wtam2018 I've added empty examples for all the resources to remove warnings. `make bundle` won't update any files in the base